kubernetes_platform/python/test/unit/test_secret.py

```diff
@@ -147,6 +147,34 @@ def my_pipeline():
             }
         }
 
+    def test_with_secret_name_param(self):
+        @dsl.pipeline
+        def my_pipeline(secret_name: str = 'my-secret'):
```

Comment:
It is missing a test that assert from the value `my-secret`

Reply:
The problem here is the pipeline spec generated from this compile step has the name stored in as a variable under the `roots` section.
```
root:
  dag:
    tasks:
      print-secret:
        cachingOptions:
          enableCache: true
        componentRef:
          name: comp-print-secret
        taskInfo:
          name: print-secret
  inputDefinitions:
    parameters:
      secret_name:
        defaultValue: my-secret
        isOptional: true
        parameterType: STRING
schemaVersion: 2.1.0
sdkVersion: kfp-2.8.0
---
platforms:
  kubernetes:
    deploymentSpec:
      executors:
        exec-print-secret:
          secretAsVolume:
          - mountPath: /mnt/my_vol
            optional: false
            secretName: secret_name
```

I tried a bunch of combinations to have the `defaultValue` field validated, so we know the name is being passed correctly, but the structure of this tests is such that if I pass in the entire pipelineSpec in the assert it fails:
https://github.com/kubeflow/pipelines/actions/runs/10191867165/job/28193820249#step:4:886

The tests only work when `platform_spec` is extracted and matched.

Note that the assert statement still verifies that the `secret_name` variable is getting passed into the `secretAsVolume`, I believe that's a good enough preliminary test at this point. Wdyt? @diegolovison @gregsheremeta

Reply:
for posterity

> The problem here is the pipeline spec generated from this compile step has the name stored in as a variable under the roots section.

we discovered today that that's not how this works. `secretName` (at the very bottom of the platforms spec) has to say `secretName: my-secret` and not `secretName: secret_name`

Reply:
also for posterity, the above comment is also incorrect :sweat_smile: 

`secretName: secret_name` actually needs to be a key to the input parameter map, so we can grab the parameter by name at runtime. Newly added changes to driver.go show how we do that.